airgap/terraform/user-data-scripts/provision_registry_server.sh.tpl

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash 
+
+set -e
+
+cp ~ubuntu/.ssh/authorized_keys ~root/.ssh/authorized_keys
+service sshd reload
+
+# (1) install docker and run registry container
+
+apt-get update
+apt-get install -y docker.io
+docker run -d -p 127.0.0.1:5000:5000 --name registry registry:2
+
+# (2) install nginx and disable the default site configuration
+
+apt install -y curl gnupg2 ca-certificates lsb-release ubuntu-keyring
+
+curl https://nginx.org/keys/nginx_signing.key | gpg --dearmor \
+    | sudo tee /usr/share/keyrings/nginx-archive-keyring.gpg >/dev/null
+
+echo "deb [signed-by=/usr/share/keyrings/nginx-archive-keyring.gpg] \
+http://nginx.org/packages/ubuntu `lsb_release -cs` nginx" \
+    | sudo tee /etc/apt/sources.list.d/nginx.list
+
+echo -e "Package: *\nPin: origin nginx.org\nPin: release o=nginx\nPin-Priority: 900\n" \
+    | sudo tee /etc/apt/preferences.d/99nginx
+
+apt update
+apt install -y nginx
+
+mv /etc/nginx/conf.d/default.conf  /etc/nginx/conf.d/default.conf.disabled
+
+# (3) install letsencrypt certbot tool to create the certificates
+
+snap install core
+snap refresh core
+snap install --classic certbot
+
+# (4) create a directory for letsencrypt challenge, and create nginx configuration for it.
+
+mkdir -p  /var/www/letsencrypt
+chown nginx:nginx -R /var/www
+
+cat << EOF > /etc/nginx/letsencrypt.conf
+location ^~ /.well-known {
+    root /var/www/letsencrypt;
+    default_type "text/plain";
+}
+EOF
+
+# (5) create a htpasswd file to be used for authentication
+
+USER=${registry_username}; PASSWORD=${registry_password}; echo "${registry_username}:$(openssl passwd -stdin -apr1 <<< ${registry_password})" > /etc/nginx/registry.password
```

Comment:
Why we need `USER=${registry_username}; PASSWORD=${registry_password};`? you already echo the generated user:password to the file.

Reply:
Refined